test/tests/test_harness/parse_errors

```diff
@@ -4,5 +4,6 @@
     input = good.i
     check_input = true
     check_input = true
+    allow_deprecated_until = 13/32/2000
```

Comment:
What's this line do?

Reply:
Allows us to test the invalid date parser:
```pre

Parser Errors:
-------------------------------------------------------------------------------------------------------------
/Users/milljm/projects/moose/test/tests/test_harness/parse_errors:5: duplicate parameter "Tests/syntax/check_input"
/Users/milljm/projects/moose/test/tests/test_harness/parse_errors:6: duplicate parameter "Tests/syntax/check_input"
/Users/milljm/projects/moose/test/tests/test_harness/parse_errors:7: invalid date value: 'Tests/syntax/allow_deprecated_until="13/32/2000"'
```

We were never testing that before...